src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -521,31 +521,38 @@ export async function getFeatureIdType(output: Log, env: NodeJS.ProcessEnv, user
 	}
 }
 
-export function getBackwardCompatibleFeatureId(id: string) {
+export function getBackwardCompatibleFeatureId(output: Log, id: string) {
 	const migratedfeatures = ['aws-cli', 'azure-cli', 'desktop-lite', 'docker-in-docker', 'docker-from-docker', 'dotnet', 'git', 'git-lfs', 'github-cli', 'java', 'kubectl-helm-minikube', 'node', 'powershell', 'python', 'ruby', 'rust', 'sshd', 'terraform'];
 	const renamedFeatures = new Map();
 	renamedFeatures.set('golang', 'go');
 	renamedFeatures.set('common', 'common-utils');
-	// TODO: Add warning log messages only when auto-mapping is ready to be put in Remote-Containers STABLE version.  
-	// const deprecatedFeatures = ['fish', 'gradle', 'homebrew', 'jupyterlab', 'maven'];
+
+	const deprecatedFeaturesIntoOptions = new Map();
+	deprecatedFeaturesIntoOptions.set('gradle', 'java');
+	deprecatedFeaturesIntoOptions.set('maven', 'java');
+	deprecatedFeaturesIntoOptions.set('jupyterlab', 'python');
+
+	// const deprecatedFeatures = ['fish', 'homebrew'];
```

Comment:
Hi @bamurtaugh, have we heard back from the maintainer of `fish` and `homebrew` on a new location for these Features ? Wondering what logs would be the best for these two Features. 

Also, could you help take a look at the wordings of these logs? Thanks

![image](https://user-images.githubusercontent.com/24955913/188984437-e72d3dc4-016c-4fe3-9bf7-e95dec5e99d6.png)

Reply:
I haven't heard anything new from him.

I like the log wording! A slight amendment idea (since users at this level aren't necessarily thinking about the spec):

> WARNING: Using the deprecated `node` Feature. See <https URL...> for the updated Feature.

Reply:
> I haven't heard anything new from him. I plan to follow up in the coming days.

Added a `TODO` comment to add suitable warnings for `fish` and `homebrew` when we have more context. I believe we shouldn't log anything until we can route the users somewhere or have some strong explanation for it's deprecation. What do you think @bamurtaugh ? 

Also, Updated the logs. thanks! 
![image](https://user-images.githubusercontent.com/24955913/188991416-59cd775d-7879-4483-98ce-45864ed05f6f.png)

Reply:
> I believe we shouldn't log anything until we can route the users somewhere or have some strong explanation for it's deprecation.

This makes sense to me - so at this time, users will be able to use the Fish and Homebrew Features as they have been, with no extra logging or disruption?

Reply:
Yes, no extra loggings for `fish` and `homebrew`